src/ts/controller/core.ts

```diff
@@ -0,0 +1,263 @@
+/**
+ * @fileoverview Core methods for schedule block placing
+ * @author NHN FE Development Lab <dl_javascript@nhn.com>
+ */
+import ScheduleViewModel from '@src/model/scheduleViewModel';
+import inArray from 'tui-code-snippet/array/inArray';
+import isUndefined from 'tui-code-snippet/type/isUndefined';
+
+import Collection, { Filter } from '@src/util/collection';
+import Schedule from '@src/model/schedule';
+import TZDate from '@src/time/date';
+import {
+  makeDateRange,
+  MILLISECONDS_PER_DAY,
+  format,
+  toStartOfDay,
+  toEndOfDay
+} from '@src/time/datetime';
+
+export type CollisionGroup = Array<number[]>;
+export type Matrix<T> = Array<Array<T[]>>;
+export type ScheduleMatrix<T> = Matrix<T>;
+export type ScheduleMatrix2d<T> = Array<T[]>;
+
+/**
+ * Calculate collision group.
+ * @param {Array<Schedule|ScheduleViewModel>} schedules List of viewmodels.
+ * @returns {Array<number[]>} Collision Group.
+ */
+export function getCollisionGroup(schedules: Schedule[] | ScheduleViewModel[]) {
+  const collisionGroups: CollisionGroup = [];
+  let previousScheduleList: Array<Schedule | ScheduleViewModel>;
+
+  if (!schedules.length) {
+    return collisionGroups;
+  }
+
+  collisionGroups[0] = [schedules[0].cid()];
+  schedules.slice(1).forEach((schedule: Schedule | ScheduleViewModel, index: number) => {
+    previousScheduleList = schedules.slice(0, index + 1).reverse();
+
+    // If overlapping previous schedules, find a Collision Group of overlapping schedules and add this schedules
+    const found = previousScheduleList.find((previous: Schedule | ScheduleViewModel) => {
+      return schedule.collidesWith(previous);
```

Comment:
If the type assigned to `previous` is a **Schedule** or **ScheduleViewModel**, the parameter type of `collidesWith ` must also be changed.

Currently only `Schedule`.

Reply:
The parameter is union type.
```js
collidesWith(schedule: Schedule | ScheduleViewModel): boolean 
```